src/js/testem-component.js

```diff
@@ -0,0 +1,531 @@
+            args:     ["http://localhost:%port", { port: "{that}.options.coveragePort" }]
+        }
+    },
+    coverageDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "coverage", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    reportsDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "reports", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    testemDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "user_data_dir", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    instrumentSource: true,
+    // Code to be served must live under the cwd in order to work with Testem.
+    instrumentedSourceDir: {
+        expander: {
+            funcName: "gpii.testem.resolveSafely",
+            args:     ["{that}.options.testemOptions.cwd", "instrumented"]
+        }
+    },
+    generateCoverageReport: true,
+    sourceDirs: [],
+    testPages:   [],
+    serveDirs:  [],
+    members: {
+        generatedOptions: {
+            routes: {}
+        }
+    },
+    events: {
+        initialCleanupComplete: null,
+        constructFixtures: null,
+        sourceInstrumented: null,
+        safeToConstructFixtures: {
+            events: {
+                initialCleanupComplete: "initialCleanupComplete",
+                constructFixtures:      "constructFixtures",
+                sourceInstrumented:     "sourceInstrumented"
+            }
+        },
+        onExpressStarted: null,
+        onFixturesConstructed: {
+            events: {
+                onExpressStarted: "onExpressStarted"
+            }
+        },
+        stopFixtures: null,
+        onExpressStopped: null,
+        onFixturesStopped: {
+            events: {
+                onExpressStopped: "onExpressStopped"
+            }
+        },
+        createReports: null,
+        onCoverageReportComplete: null,
+        onReportsComplete: {
+            events: {
+                onCoverageReportComplete: "onCoverageReportComplete"
+            }
+        },
+        finalCleanup: null,
+        finalCleanupComplete: null
+    },
+    testemOptions: {
+        framework:   "qunit",
+        report_file: {
+            expander: {
+                funcName: "gpii.testem.resolveSafely",
+                args:     ["{that}.options.reportsDir", "report.tap"]
+            }
+        },
+        cwd: process.cwd(),
+        user_data_dir: "{that}.options.testemDir",
+        on_start: "{that}.handleTestemStart",
+        on_exit:  "{that}.handleTestemExit",
+        test_page: "{that}.options.testPages",
+        proxies: {
+            "/coverage": {
+                "target": "{that}.options.coverageUrl"
+            }
+        }
+    },
+    invokers: {
+        "handleTestemStart": {
```

Comment:
I think that someone familiar with Infusion's idioms will still find the design of these functions puzzling and unidiomatic. Especially, "invoker that registers event listener" to me smells like a bit of an antipattern. The point of registering listeners declaratively is in order to make the design of an application clearer to authors - especially given that future tooling will expose this structure usefully.

The design of Infusion components actively promote ways in which misconfiguration may cause fundamental errors, at least in the opinion of their original author. I'd like to go one more round of promoting my original suggestion of implementing these functions as namespaced listeners plus a comment warning the user that overriding the one that triggers the callback without substituting an equivalent effect will cause a hung process. But this is something we can certainly discuss at a 1-to-1 next week - is your traditional Monday afternoon slot available? I can also do mornings now :)